### PR TITLE
{re3,reVC,reLCS}: init at unstable-2021-09-03

### PR DESCRIPTION
**Pull request by luc65r**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by luc65r** [comment]
Needs #129367 to be merged

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130089` at 9037e68e run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>embree</li>
    <li>embree2</li>
    <li>forge</li>
    <li>luxcorerender</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/df8d4b43498edef7c23bb25d627b357f">ccextractor</a></li>
    <li><a href="https://gist.github.com/8317bfeb7f38017ecbea7759910c21dc">vcv-rack</a></li>
  </ul>
</details>
<details>
  <summary>4 packages skipped due to time constraints:</summary>
  <ul>
    <li>MIDIVisualizer</li>
    <li>idrisPackages.glfw</li>
    <li>tev</li>
    <li>tracy</li>
  </ul>
</details>
<details>
  <summary>23 packages built successfully:</summary>
  <ul>
    <li>glfw (glfw3)</li>
    <li>glfw-wayland</li>
    <li>glslviewer</li>
    <li>goxel</li>
    <li>librw</li>
    <li>librw-wayland</li>
    <li>libsForQt5.mapbox-gl-native (libsForQt515.mapbox-gl-native ,plasma5Packages.mapbox-gl-native)</li>
    <li>libsForQt5.mapbox-gl-qml (libsForQt515.mapbox-gl-qml ,plasma5Packages.mapbox-gl-qml)</li>
    <li>libsForQt512.mapbox-gl-native</li>
    <li>libsForQt512.mapbox-gl-qml</li>
    <li>libsForQt514.mapbox-gl-native</li>
    <li>libsForQt514.mapbox-gl-qml</li>
    <li>pixelnuke</li>
    <li>pure-maps</li>
    <li>raylib</li>
    <li>re3</li>
    <li>re3-wayland</li>
    <li>relcs</li>
    <li>relcs-wayland</li>
    <li>revc</li>
    <li>revc-wayland</li>
    <li>taisei</li>
    <li>vlang</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/libraries/librw/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+, lib
+, fetchFromGitHub
+, glfw
+, cmake
+, withTools ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "librw";
+  version = "09-07-2021";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/librw/default.nix:11]:
```suggestion
  version = "unstable-09-07-2021";
```

File: pkgs/games/re3/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    };
+  };
+
+  lcs = callPackage ./base.nix rec {
+    pname = "reLCS";
+    version = "12-07-2021";
+
+    src = fetchFromGitHub {
+      repo = "re3";
+      owner = "GTAmodding";
```
Comment (SuperSandro2000) [inline pkgs/games/re3/default.nix:37]:
```suggestion
      owner = "GTAmodding";
      repo = "re3";
      rev = "a4b92fe9bec17ad00dc436574f57844c4b8d49f5";
      sha256 = "5CPgDhUgkB0B5EApwkdhh7qqEbtCE3Fee1CKu3BTGz4=";
    };
  };

  miami = callPackage ./base.nix rec {
    pname = "reVC";
    version = "12-07-2021";

    src = fetchFromGitHub {
      owner = "GTAmodding";
      repo = "re3";
      rev = "523b23339cd761d934501dcbcc059d131b5a28c4";
      sha256 = "Cb9C9Gh1VGPPwF2uO82wOdxD2rfaixdkBsClx3OFNvU=";
    };
  };

  lcs = callPackage ./base.nix rec {
    pname = "reLCS";
    version = "12-07-2021";

    src = fetchFromGitHub {
      owner = "GTAmodding";
      repo = "re3";
```

File: pkgs/games/re3/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{ branch ? "re3"
+, callPackage
+, fetchFromGitHub
+}:
+
+{
+  re3 = callPackage ./base.nix rec {
+    pname = "re3";
+    version = "12-07-2021";
```
Comment (SuperSandro2000) [inline pkgs/games/re3/default.nix:9]:
```suggestion
    version = "unstable-12-07-2021";
```
Why are we not using the same source for all packages?

File: pkgs/games/re3/default.nix (hunk 2 above)
Comment (luc65r) [reply pkgs/games/re3/default.nix]: Is that serious? Just exchanging owner and repo lines?

Comment (SuperSandro2000) [reply pkgs/games/re3/default.nix]: Yes, it should have this order to make it easier to find the repo.

File: pkgs/games/re3/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  lcs = callPackage ./base.nix rec {
+    pname = "reLCS";
+    version = "unstable-2021-07-12";
+
+    src = fetchFromGitHub {
+      repo = "re3";
+      owner = "GTAmodding";
+      rev = "6f0f9d7d695056575120ff49cc01297419c0d0e4";
```
Comment (SuperSandro2000) [inline pkgs/games/re3/default.nix:38]: Can we use the same rev for all packages?

Comment (luc65r) [reply pkgs/games/re3/default.nix:38]: No, different branches mean different commits

File: pkgs/games/re3/default.nix (hunk 2 above)
Comment (luc65r) [reply pkgs/games/re3/default.nix]: Ok, fine

File: pkgs/games/re3/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/games/re3/default.nix:38]: Can you add the branch information in a comment before the rev?

Comment (luc65r) [reply pkgs/games/re3/default.nix]: Now I think this is clearer with the real branch names

File: pkgs/games/re3/base.nix [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      find_package(X11 REQUIRED)
+      target_include_directories(\''${EXECUTABLE} PRIVATE \''${X11_INCLUDE_DIR})
+      target_link_libraries(\''${EXECUTABLE} PRIVATE \''${X11_LIBRARIES})
+    EOF
+  '';
+
+  meta = with lib; {
+    description = "An open-source project reverse-engineering Grand Theft Auto";
+    homepage = "https://github.com/GTAmodding/re3";
+    license = licenses.free;
```
Comment (veprbl) [inline pkgs/games/re3/base.nix:64]:
```suggestion
    license = licenses.unfree;
```

Comment (luc65r) [reply pkgs/games/re3/base.nix:64]: > We don't feel like we're in a position to give this code a license.
> The code should only be used for educational, documentation and modding purposes.
> We do not encourage piracy or commercial use.
> Please keep derivate work open source and give proper credit.

that is very imprecise, but I feel like this is free software

**Comment by luc65r** [comment]
The repository has been (once again) DMCA'd

**Pull request closed without merging**